Jenkinsfile-downloads

```diff
@@ -0,0 +1,43 @@
+import groovy.json.JsonSlurper
+
+pipeline{
+	agent any
+
+	stages{
+		stage('Check upstream builds succeeded'){
+			steps{
+				script{
+					def configStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/ConfirmReleaseConfigs/lastBuild/api/json"
+					def configStatusJson = new JsonSlurper().parseText(configStatusUrl.getContent())
+					if(configStatusJson['result'] != "SUCCESS"){
+						error("Most recent ConfirmReleaseConfigs build status: " + configStatusJson['result'] + ". Please complete a successful build.")
+					}
+			    	}	
+		    	}
+	    	}
+        	stage('Setup: Build jar file'){
+            		steps{
+	    			script{
+                    			dir('add-links'){
+						sh "mvn clean package -DskipTests"
+                    			}
+          	    		}
+            		}		
+        	}
+		stage('Main: Download AddLinks files'){
+			steps{
+				script{
+					dir('add-links'){
+						withCredentials([file(credentialsId: 'Config', variable: 'FILE')]){
+							sh "cp $FILE src/main/resources/auth.properties"
+							sh "ln -sf src/main/resources resources"
+							sh "rm -f src/main/resources/db.properties"
```

Comment:
Why is `db.properties` removed here?  Is it handled by Jenkins master config file?

Reply:
The addlinks process basically searches in the resources folder for *.properties, and then creates a kind of mega properties file to be used by the program. `db.properties` is a default file that has most of its attributes blank. With Jenkins now supplying the config file, the path of least resistance was to have Jenkins replace one of the two .properties files (auth.properties or db.properties), and then remove the other file so that duplicate key-value pairs don't pollute the process